<a href="https://colab.research.google.com/github/SuastrianDika/demo/blob/master/Hypothesis_Testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

####PT Maju Makmur Sentosa Raya ingin mengetahui apakah desain web baru mereka dapat meningkatkan conversion rate web mereka. Perusahaan tersebut memiliki data historis conversion rate dari desain lama dan desain baru selama 30 hari ke belakang. Data tersebut diberikan dalam format csv dengan kolom conversion_a adalah conversion rate harian dari desain web lama dan kolom conversion_b adalah conversion rate harian dari desain web baru. Perusahaan ingin mengetahui apakah rata-rata conversion rate dari desain baru lebih besar dari desain lama.

####1.Apa hipotesis null dan hipotesis alternatif dari kasus tersebut?

####(H0: μ1 = μ2) Rata-rata conversion rate dari desain baru tidak lebih besar dari desain lama.

####(H1: μ1 < μ2) Rata-rata conversion rate dari desain baru lebih besar dari desain lama.

In [ ]:
import pandas as pd

df = pd.read_csv('https://raw.githubusercontent.com/SuastrianDika/dataset/main/hw_data.csv')

df.head()

Day   Conversion_A  Conversion_B
0     1          0.15          0.19
1     2          0.12          0.20
2     3          0.18          0.18
3     4          0.19          0.22
4     5          0.17          0.19

In [ ]:
df.describe()

Day   Conversion_A  Conversion_B
count  30.000000     30.000000     30.000000
mean   15.500000      0.162000      0.193333
std     8.803408      0.036141      0.027334
min     1.000000      0.100000      0.150000
25%     8.250000      0.132500      0.180000
50%    15.500000      0.165000      0.190000
75%    22.750000      0.187500      0.200000
max    30.000000      0.230000      0.270000

####2. Apa uji hipotesis yang dapat kita lakukan?
####Two-sample paired Z-test for Means.

In [ ]:
from statsmodels.stats.weightstats import ztest as ztest
    
A = df['Conversion_A']
B = df['Conversion_B']

z, p_value = ztest(A, B, alternative='larger', value=0)

In [ ]:
print(z.round(3), p_value.round(4))

-3.787 0.9999


####4.Apa kesimpulan yang dapat ditarik dari uji tersebut?

In [ ]:
if p_value < 0.05:
  print("Null hyphothesis ditolak , Alternative hyphothesis diterima")
else:
  print("Null hyphothesis diterima, Alternative hyphothesis ditolak")

Null hyphothesis diterima, Alternative hyphothesis ditolak


####Kesimpulan :
####Daerah Penolakan H0 untuk hipotesis one-tailed kanan(H0 : Z > Zα) dimana Z = -3.787 dan Zα = 1.65 tidak terpenuhi. 
####Rata-rata conversion rate dari desain baru tidak lebih besar dari desain lama.